plugins/ltac/tacintern.ml

```diff
@@ -95,9 +95,16 @@ let intern_string_or_var = intern_or_var (fun (s : string) -> s)
 let intern_global_reference ist qid =
```

Comment:
After pondering, I prefer keeping it that way. There are many other similarly named functions in this module, which would add to the confusion, and furthermore, goal variables are actually global variables in so far as they correspond to the `VarRef` node.

Reply:
> as they correspond to the `VarRef` node.

It depends what meaning you give to the `VarRef` node. For instance, goal variables have no entry in the nametab, and in this sense they are `Var` and not `VarRef` node.

Reply:
Right, but it's still open to interpretation. And as I said, you also have in the same file:
- `intern_constr_reference`
- `intern_isolated_global_tactic_reference`
- `intern_isolated_tactic_reference`
- `intern_applied_global_tactic_reference`
- `intern_applied_tactic_reference`
- `intern_non_tactic_reference`
- `intern_evaluable_global_reference`

Reply:
Then, it should be called `intern_constr_reference` and:
- either the current `intern_constr_reference` should be renamed to `intern_global_constr_reference` (since it is what it actually does);
- or it should be relaxed and be just the embedding of the result of your new function as a `glob_constr`, optional `constr_expr` pair.
This would be more consistent to me, wouldn't it?

Reply:
I really don't want to touch this more than necessary. It's intricate code and it's easy to mess it up without realizing...

Reply:
I can indeed understand your view!

What makes me afraid is that if, one day, I succeed to make this code less intricate than it is, will you be supportive? I think you know my view which is that the better we understand and clarify the Ltac implementation, the easier a mechanical transition to Ltac2 will be possible. But I can hardly work in this direction without a minimal support.

Reply:
The two things I would complain about are 1. more intricate implementations 2. backwards compatibility. Given the way Ltac1 is defined, it's fairly hard to respect 2., so I am inclined to tolerate *minor* semantic changes, i.e. stuff that requires very few changes to the devs tracked in our CI. The one thing we can't do is pull the rug under the feet of our users.

That said, I also don't believe there is any point in making Ltac1 converge towards Ltac2. The "simple" stuff is already straightforward to port to Ltac2, everything else is bonkers, and trying to change it in Ltac1 is just going to break everything. I won't deter you from doing it, and I have a hard time understanding why you believe that there is such a thing as "the one tactic language of Coq". By contrast, I am supportive of the development of several, very different tactic languages, e.g. Mtac2, coq-elpi, etc. As long as we have an FFI, why should they look the same, since they don't even pertain to the same family of languages!?

Reply:
I'm ok with several tactic languages. I'm just trying to understand where we are going. About what we are considering to be technical debt and what we are considering simply as diversity. In particular, if adding new tactics, new introduction patterns, smoothing the semantics of Ltac1 etc. is fine for you, there is no problem for me, and I'll feel free to come back on those tactic PRs about which I felt a resistance from you.